pkg/mixer/lint.go

```diff
@@ -101,34 +107,45 @@ var sentenceRegexp = regexp.MustCompile(`^[A-Z].+\.$`)
 
 // Enforces alerting guidelines.
 // https://monitoring.mixins.dev/#guidelines-for-alert-names-labels-and-annotations
-func lintPrometheusAlertsGuidelines(rule *rulefmt.RuleNode) (errs []error) {
-
-	if !camelCaseRegexp.MatchString(rule.Alert.Value) {
-		errs = append(errs, fmt.Errorf("[alert-name-camelcase] Alert '%s' name is not in camel case", rule.Alert.Value))
+func lintPrometheusAlertsGuidelines(rule *rulefmt.RuleNode, cf *lint.ConfigurationFile) (errs []error) {
```

Comment:
Nit, also not necessary to refactor in this PR, but something to note.

This function is getting increasingly complex, with lots of `!=` conditionals and nested conditionals. A refactor which separates rules into their own functions, and simplifies the logic would be welcome in the future.

Reply:
Completely agree, it's definitely not a long term solution as-is. 

Ideally I'd like to refactor the configuration to be more extensible and repackage it as an independent module separate from both mixtool and dashboard-linter, and then update the alert rules to follow the dashboard-linter approach.